### PR TITLE
Api testing

### PR DESCRIPTION
**Pull request by perryr16**
## Generated tests for existing routes 
* to set up rspec was a challenge
* in order to run the app, `$ rackup config.ru`
* now on localhost:9292
* attempt to hide API key

### CODE REVIEW

File: config.ru [hunk 1]
```diff
@@ -0,0 +1,2 @@
+require './i-scream-arcade-microservice-app'
+run IScreamMicroservice
```
Comment (perryr16) [inline config.ru:2]: This is the file that runs the microservice, needed to run rspec (at least this is what worked for me)

File: config/environment.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+require "rubygems"
+require "bundler"
+Bundler.require(:default)                   # load all the default gems
+Bundler.require(Sinatra::Base.environment)  # load all the environment specific gems
+require "active_support/deprecation"
+require "active_support/all"
+$db = []
```
Comment (perryr16) [inline config/environment.rb:7]: Found on the internet, again, allowed me to run rspec

File: i-scream-arcade-microservice-app.rb [hunk 3]
```diff
@@ -1 +1,46 @@
...
+require 'sinatra/json'
+require 'json'
+require 'faraday'
+require 'pry'
+require './services/game_service'
+require './poros/game_results'
+
+
+
+class IScreamMicroservice < Sinatra::Base
```
Comment (perryr16) [inline i-scream-arcade-microservice-app.rb:11]: had to put all the code in a Sinatra::Base class

File: spec/exact_game_match_spec.rb [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+RSpec.describe IScreamMicroservice do
+  include Rack::Test::Methods
+  def app
+    IScreamMicroservice.new # this defines the active application for this test
+  end
+
+  it "test" do
+    get '/dog'
+
+    expect(last_response.body).to include("what up dog")
```
Comment (perryr16) [inline spec/exact_game_match_spec.rb:14]: `last_response` is a built in method that returns exactly what it sounds like

**Review by perryr16 — COMMENTED** [review]
Many relevant comments in the PR "API Building", Also, Travis CI has not been setup for this project. Probably something we should do

File: poros/game_results.rb [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    array.map do |element|
+      service.send(method,element)[0][key]
+    end
+  end
+
+  def cover_url(cover_id)
+    service.get_cover(cover_id)[0][:url] # id
+  end
+  
+  def youtube_id(video_id)
```
Comment (melatran) [inline poros/game_results.rb:39]: Kind of curious if we even need a youtube api or could we just use the embedded code and pass the youtube video id from the api in here. This currently goes to the kissing booth 2 but we could just pass whatever video code the api returns and pass in here in our app.

```
<iframe width="560" height="315" src="https://www.youtube.com/embed/**fjVonI2oVeM**" frameborder="0" 
allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
```

File: spec/exact_game_match_spec.rb [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    expect(last_response.body).to include(exp_screenshots)
+    expect(last_response.body).to include(exp_themes)
+    expect(last_response.body).to include(exp_videos)
+
+
+  end
+  
+  
+
+end
```
Comment (melatran) [inline spec/exact_game_match_spec.rb:79]: Tests look great

File: config.ru (hunk 1 above)
Comment (melatran) [reply config.ru:2]: Cool. That's good to know

File: Gemfile [hunk 7]
```diff
@@ -1,5 +1,20 @@
...
+gem 'rack'
+gem 'sinatra', require: 'sinatra/base'
+gem 'sinatra-contrib'
 gem 'figaro'
 gem 'faraday'
 gem 'pry'
-gem 'rspec'
+gem 'json'
+gem 'fast_jsonapi'
+gem 'shotgun'
```
Comment (melatran) [inline Gemfile:11]: **Note for myself:** The shotgun command starts one of Rack's supported servers (e.g., mongrel, thin, webrick) and listens for requests but does not load any part of the actual application. Each time a request is received, it forks, loads the application in the child process, processes the request, and exits the child process. The result is clean, application-wide reloading of all source files and templates on each request. _Shotgun only works on Mac and Linux_

https://learn.co/lessons/sinatra-shotgun-server

File: config.ru [hunk 8]
```diff
@@ -0,0 +1,2 @@
+require './i-scream-arcade-microservice-app'
```
Comment (melatran) [inline config.ru:1]: **When to use a config.ru?** A config.ru file is recommended if:

_You want to deploy with a different Rack handler (Passenger, Unicorn, Heroku, …)._
You want to use more than one subclass of Sinatra::Base.
You want to use Sinatra only for middleware, and not as an endpoint.

File: spec/exact_game_match_spec.rb [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+    expect(last_response.body).to include(exp_total_rating)
+    expect(last_response.body).to include(exp_categories)
+    expect(last_response.body).to include(exp_genres)
+    expect(last_response.body).to include(exp_keywords)
+    expect(last_response.body).to include(exp_platforms)
+    expect(last_response.body).to include(exp_similars)
+    expect(last_response.body).to include(exp_screenshots)
+    expect(last_response.body).to include(exp_themes)
+    expect(last_response.body).to include(exp_videos)
+
```
Comment (Mariana-21) [inline spec/exact_game_match_spec.rb:73]: Crazy testing!!!

**Pull request merged**